src/app/qgisapp.cpp

```diff
@@ -630,6 +630,7 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipVersionCh
   }
 
   sInstance = this;
+  Qgis::ogrRuntimeSupport();
```

Comment:
Based on another error situation done further down, I added this: 
<pre>
 if (!Qgis::ogrRuntimeSupport())
  {
    std::cerr << QObject::tr(
                QString("QGIS does not support this version of GDAL.\n"
                "GDAL[%1]\n"
                "Please install a GDAL 2 version.\n").arg(Qgis::GDAL_RUNTIME_VERSION)
              ).toUtf8().constData();
    exit( 1 ); 
  }
</pre>